weed/server/master_grpc_server.go

```diff
@@ -18,6 +22,41 @@ import (
 	"github.com/chrislusf/seaweedfs/weed/topology"
 )
 
+func (ms *MasterServer) RegisterUuids(heartbeat *master_pb.Heartbeat) (duplicated_uuids []string, err error) {
+	ms.Topo.UuidAccessLock.Lock()
+	defer ms.Topo.UuidAccessLock.Unlock()
+	key := fmt.Sprintf("%s:%d", heartbeat.Ip, heartbeat.Port)
+	if ms.Topo.UuidMap == nil {
+		ms.Topo.UuidMap = make(map[string][]string)
+	}
+	// find whether new uuid exists
+	for k, v := range ms.Topo.UuidMap {
```

Comment:
just need to check whether the uuid is in the uuidMap. The double loop is unnecessary.

Reply:
I think it's necessary. 
`UuidMap        map[string][]string` UuidMap is a map of slice. `LocationUuids   []string ` is also a slice.
I need to check whether uuid in heartbeat.LocationUuids exists in every slice in value of UuidMap .

For example:  
UuidMap = {"10.10.10.11:1111":[uuid1 uuid2 uuid3],     "10.10.10.12:1112":[uuid4 uuid5 uuid6],}
heartbeat.LocationUuids = {"10.10.10.11:1113":[uuid1 uuid4 uuid7], }
I need 2 loops to check whether uuid1 uuid4 and uuid7 exist in UuidMap.

Or maybe I don't really get your opinions, if I was wrong , please point it out.